data-report-commons/src/main/java/commons/formatter/CsvFormatter.java

```diff
@@ -1,17 +1,116 @@
 package commons.formatter;
 
-import java.io.ByteArrayOutputStream;
+import static org.apache.commons.io.StandardLineSeparator.CRLF;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.List;
 import nva.commons.core.JacocoGenerated;
+import org.apache.jena.query.QuerySolution;
 import org.apache.jena.query.ResultSet;
-import org.apache.jena.query.ResultSetFormatter;
+import org.apache.jena.rdf.model.Literal;
+import org.apache.jena.rdf.model.RDFNode;
 
 @JacocoGenerated //Tested in modules report-api and bulk-export
 public class CsvFormatter implements ResponseFormatter {
 
+    private static final String DECIMAL_PATTERN = "0.0000";
+    private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat(DECIMAL_PATTERN,
+                                                                          DecimalFormatSymbols.getInstance());
+    private static final String DELIMITER = ",";
+    private static final String TYPE_INTEGER = "http://www.w3.org/2001/XMLSchema#integer";
+    private static final String TYPE_DOUBLE = "http://www.w3.org/2001/XMLSchema#double";
+    private static final String LINE_BREAK = CRLF.getString();
+
     @Override
     public String format(ResultSet resultSet) {
-        var outputStream = new ByteArrayOutputStream();
-        ResultSetFormatter.outputAsCSV(outputStream, resultSet);
-        return outputStream.toString();
+        var resultVars = resultSet.getResultVars();
```

Comment:
Did not want to replace the elegant code
```
var outputStream = new ByteArrayOutputStream();
ResultSetFormatter.outputAsCSV(outputStream, resultSet);
return outputStream.toString();
```
But double values where formatted as "4.5961E0" instead of "4.5961". Feel free to suggest better way to format this.